fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+    if (ret != 0)
+    {
+        SWSS_LOG_INFO("Failed cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+    }
+
+    SWSS_LOG_NOTICE("Success cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+
+    return;
+}
+
+void FdbSync::macUpdateStateDB(struct fdb_info *info)
+{
+    char *op;
+    char *type;
+    string port_name = "";
+    string key = "Vlan" + to_string(info->vid) + ":" + info->mac;
+	short fdb_type;    /*dynamic or static*/
+
+    if (info->op_type == FDB_OPER_ADD)
+    {
+        macAddSrcDB(info);
+        op = "add";
+        port_name+= info->port_name;
+        fdb_type = info->type;
+        /* Check if this vlan+key is also learned by vxlan neighbor then delete learned on */
+        if (m_mac.find(key) != m_mac.end())
+        {
+            op = "replace";
+            macDelVxlanEntry(key, info);
+            SWSS_LOG_INFO("Local learn event deleting from VXLAN table DEL_KEY %s", key.c_str());
+            macDelVxlan(key);
+        }
+    }
+    else
+    {
+        op = "del";
+        if (macCheckSrcDB(info) == false)
+        {
+            return;
+        }
+		port_name+= fdbmac[key].port_name;
+		fdb_type = fdbmac[key].type;
+		fdbmac.erase(key);
+    }
+
+    if (fdb_type == FDB_TYPE_DYNAMIC)
+    {
+        type = "dynamic";
+    }
+    else
+    {
+        type = "static";
+    }
+
+    const std::string cmds = std::string("")
+        + " bridge fdb " + op + " " + info->mac + " dev " 
+        + port_name + " master " + type + " vlan " + to_string(info->vid);
+
+    std::string res;
+    int ret = swss::exec(cmds, res);
+
+    SWSS_LOG_NOTICE("cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+
+    return;
+}
+
+void FdbSync::macRefreshStateDB(int vlan, string kmac)
+{
+    string key = "Vlan" + to_string(vlan) + ":" + kmac;
+    char *type;
+    string port_name = "";
+
+	SWSS_LOG_NOTICE("Refreshing Vlan:%d MAC route MAC:%s Key %s", vlan, kmac.c_str(), key.c_str());
+
+    if (fdbmac.find(key)!=fdbmac.end())
+    {
+		port_name+= fdbmac[key].port_name;
+        if (port_name.empty())
+        {
+            SWSS_LOG_INFO("Port name not present MAC route Key:%s", key.c_str());
+            return;
+        }
+
+		SWSS_LOG_INFO("Refreshing MAC route MAC:%s Key %s Port %s", kmac.c_str(), key.c_str(), port_name.c_str());
+
+		if (fdbmac[key].type == FDB_TYPE_DYNAMIC)
+		{
+			type = "dynamic";
+		}
+		else
+		{
+			type = "static";
+		}
+
+		const std::string cmds = std::string("")
+			+ " bridge fdb " + "add" + " " + kmac + " dev "
+			+ port_name + " master " + type  + " vlan " + to_string(vlan);
+
+		std::string res;
+		int ret = swss::exec(cmds, res);
+		if (ret != 0)
+		{
+			SWSS_LOG_INFO("Failed cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+		}
+
+		SWSS_LOG_NOTICE("Refreshing cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+    }
+    return;
+}
+
+bool FdbSync::checkImetExit(string key, uint32_t vni)
+{
+    if (imet_route[key].redist_add_status == true)

```

Comment:
What if key doesn't exist? It will add an entry. Doesn't appear to be doing its purpose as function name